### PR TITLE
Fix MDDropdownmenu positoning issue in MD ToolBar.

### LINKED ISSUE #1108: Header Menu not working with MDToolbar
### Description of the Bug

Hi,
I try to combine the Header Menu with the Menu with MDToolbar example but the header part is not in the correct position.

### Code and Logs

```python
from kivy.lang import Builder
from kivy.metrics import dp

from kivymd.app import MDApp
from kivymd.uix.menu import MDDropdownMenu
from kivymd.uix.snackbar import Snackbar
from kivymd.uix.boxlayout import MDBoxLayout

KV = '''
<MenuHeader>
    orientation: "vertical"
    adaptive_size: True
    padding: "4dp"

    MDBoxLayout:
        spacing: "120dp"
        adaptive_size: True

        MDIconButton:
            icon: "gesture-tap-button"
            pos_hint: {"center_y": .5}

        MDLabel:
            text: "Actions"
            adaptive_size: True
            pos_hint: {"center_y": .5}
            
MDBoxLayout:
    orientation: "vertical"

    MDToolbar:
        title: "MDToolbar"
        left_action_items: [["menu", lambda x: app.callback(x)]]
        right_action_items: [["dots-vertical", lambda x: app.callback(x)]]

    MDLabel:
        text: "Content"
        halign: "center"
'''

class MenuHeader(MDBoxLayout):
    '''An instance of the class that will be added to the menu header.'''

class Test(MDApp):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        self.screen = Builder.load_string(KV)
        menu_items = [
            {
                "viewclass": "OneLineListItem",
                "text": f"Item {i}",
                "height": dp(56),
                "on_release": lambda x=f"Item {i}": self.menu_callback(x),
             } for i in range(5)
        ]
        self.menu = MDDropdownMenu(
            header_cls=MenuHeader(),
            items=menu_items,
            width_mult=4,
        )
        
    def build(self):
        return self.screen

    def callback(self, button):
        self.menu.caller = button
        self.menu.open()

    def menu_callback(self, text_item):
        self.menu.dismiss()
        Snackbar(text=text_item).open()

Test().run()
```

### Screenshots

grafik

### Versions

* OS: Win 10
* Python: Python 3.8.8
* Kivy: 2.0.0
* KivyMD: 0.104.2, git-bc7d1f5, 2021-06-06

### PR DESCRIPTION
**Pull request by sollarp**
Fix #1108 Header Menu not working with MDToolbar bug.

### Description of the problem

Drop down menu takes the window y axis which is 0 but the action item is lower on y than that. Because of this issue left and right drop down menu ends up in different position in window. When  right_action_items called in MDToolBar the  dropdown menu position not same.

#Reproducing issue
 Left icon clicked drop down menu get correct pos but when right icon called drop down menu ends up in right up corner

Minimal code example that reproduces the problem:

          from kivy.lang import Builder
          from kivy.metrics import dp
          
          from kivymd.app import MDApp
          from kivymd.uix.menu import MDDropdownMenu
          from kivymd.uix.snackbar import Snackbar
          from kivymd.uix.boxlayout import MDBoxLayout
          
          KV = '''
          <MenuHeader>
              orientation: "vertical"
              adaptive_size: True
              padding: "4dp"
          
              MDBoxLayout:
                  spacing: "120dp"
                  adaptive_size: True
          
                  MDIconButton:
                      icon: "gesture-tap-button"
                      pos_hint: {"center_y": .5}
          
                  MDLabel:
                      text: "Actions"
                      adaptive_size: True
                      pos_hint: {"center_y": .5}
                      
          MDBoxLayout:
              orientation: "vertical"
          
              MDToolbar:
                  title: "MDToolbar"
                  left_action_items: [["menu", lambda x: app.callback(x)]]
                  right_action_items: [["dots-vertical", lambda x: app.callback(x)]]
          
              MDLabel:
                  text: "Content"
                  halign: "center"
              MDBottomAppBar:
          
                  MDToolbar:
                      title: "Title"
                      icon: "git"
                      type: "bottom"
                      left_action_items: [["menu", lambda x: app.callback(x)]]
                      right_action_items: [["dots-vertical", lambda x: app.callback(x)]]
          '''
          
          class MenuHeader(MDBoxLayout):
              '''An instance of the class that will be added to the menu header.'''
          
          class Test(MDApp):
              def __init__(self, **kwargs):
                  super().__init__(**kwargs)
                  self.screen = Builder.load_string(KV)
                  menu_items = [
                      {
                          "viewclass": "OneLineListItem",
                          "text": f"Item {i}",
                          "height": dp(56),
                          "on_release": lambda x=f"Item {i}": self.menu_callback(x),
                       } for i in range(5)
                  ]
                  self.menu = MDDropdownMenu(
                      header_cls=MenuHeader(),
                      items=menu_items,
                      width_mult=4,
                  )
                  
              def build(self):
                  return self.screen
          
              def callback(self, button):
                  self.menu.caller = button
                  self.menu.open()
          
              def menu_callback(self, text_item):
                  self.menu.dismiss()
                  Snackbar(text=text_item).open()
          
       Test().run()

Screenshots of the problem

mdtoolpic
 
Describe what changes you create to solve the problem.

Took the window starting point and drop down the menu position from top - 50 also repositioned dropdown menu in window.
Position parameter only work in default (auto) any other position (center, top , bottom ) works as it was working before.
This includes  BottomAppBar.  Added shift effect to improve experience.

### Screenshots of the solution to the problem

https://user-images.githubusercontent.com/74240451/139330145-c7c4947a-934f-4cc6-aedd-4b8eae0fd394.mp4

### CODE REVIEW

**Comment by sollarp** [comment]
@HeaTTheatR 
Hope I got this PR right I have not tested this on Android but should work.

Thanks for your patient.

**Comment by HeaTTheatR** [comment]
@sollarp Your changes are not correct. They just offset the position of the menu by 100 pixels without considering the position of the parent widget. You should test your changes against examples of existing documentation to make sure your changes work correctly and do not break anything.

Снимок экрана 2021-10-29 в 07 16 53

**Comment by sollarp** [comment]
I can see where the problem is. I look into it and will get back.

File: kivymd/uix/menu/menu.py [hunk 1]
```diff
@@ -464,10 +464,11 @@ def build(self):
 __all__ = ("MDDropdownMenu",)
 
 import os
+from traceback import print_tb
 from typing import NoReturn, Union
-
```
Comment (HeaTTheatR) [inline kivymd/uix/menu/menu.py:468]: For what reason did you remove the newline separating the import libraries?

File: kivymd/uix/menu/menu.py [hunk 2]
```diff
@@ -493,7 +494,6 @@ def build(self):
 ) as kv_file:
     Builder.load_string(kv_file.read())
 
-
```
Comment (HeaTTheatR) [inline kivymd/uix/menu/menu.py]: For what reason did you remove the newline?
PEP8 require classes to be separated by two indents.

File: kivymd/uix/menu/menu.py [hunk 3]
```diff
@@ -691,13 +684,22 @@ def set_menu_properties(self, interval: Union[int, float] = 0) -> NoReturn:
...
+            self.target_height =  0 - dp(self.h_cls)
 
             # Set the target_height of the menu depending on the size of
             # each MDMenuItem or MDMenuItemIcon.
-            self.target_height = 0
             for item in self.ids.md_menu.data:
-                self.target_height += item.get("height", dp(72))
-
+                self.target_height += item.get("height", dp(48))
+
```
Comment (HeaTTheatR) [inline kivymd/uix/menu/menu.py]: Why are there extra spaces in the code?

File: kivymd/uix/menu/menu.py [hunk 4]
```diff
@@ -774,27 +776,49 @@ def set_menu_properties(self, interval: Union[int, float] = 0) -> NoReturn:
                         )
 
             if ver_growth == "down":
-                self.tar_y = self._start_coords[1] - self.target_height
+                self.tar_y = self._start_coords[1] - self.target_height - dp(self.h_cls)
             else:  # should always be "up"
                 self.tar_y = self._start_coords[1]
-
+
```
Comment (HeaTTheatR) [inline kivymd/uix/menu/menu.py]: Why are there extra spaces in the code?

File: kivymd/uix/menu/menu.py [hunk 5]
```diff
@@ -843,19 +867,22 @@ def open(interval):
...
 
     def on_header_cls(
         self, instance_dropdown_menu, instance_user_menu_header
     ) -> NoReturn:
+        
         def add_content_header_cls(interval):
             self.ids.content_header.clear_widgets()
             self.ids.content_header.add_widget(instance_user_menu_header)
-
+
```
Comment (HeaTTheatR) [inline kivymd/uix/menu/menu.py]: Why are there extra spaces in the code?

**Comment by HeaTTheatR** [comment]
@sollarp Before pushing your changes, use the command `pre-commit run --all-files`

**Comment by sollarp** [comment]
@HeaTTheatR you are absolutely right. I will correct this. Thank you for the advise.

File: kivymd/theming.py [hunk 6]
```diff
@@ -480,9 +480,8 @@ def _get_accent_dark(self) -> list:
     .. versionadded:: 1.0.0
 
     .. seealso::
-    
-       `Material Design 2 <https://material.io/>`_ and
-       `Material Design 3 <https://m3.material.io>`_
+    `Material Design 2 <https://material.io/>`_ and
+    `Material Design 3 <https://m3.material.io>`_
```
Comment (HeaTTheatR) [inline kivymd/theming.py]: Removing four spaces is a bug.

File: kivymd/uix/datatables/datatables.py [hunk 7]
```diff
@@ -1383,13 +1380,12 @@ def build(self):
...
-            ...,
-            background_color_header=get_color_from_hex("#65275d"),
-            background_color_cell=get_color_from_hex("#451938"),
+        ...,
+        background_color_header=get_color_from_hex("#65275d"),
+        background_color_cell=get_color_from_hex("#451938"),
         )
-        
     .. image:: https://github.com/HeaTTheatR/KivyMD-data/raw/master/gallery/kivymddoc/data-tables-background-color-cell.png
-        :align: center
```
Comment (HeaTTheatR) [inline kivymd/uix/datatables/datatables.py:1392]: Removing four spaces is a bug.

File: kivymd/uix/datatables/datatables.py [hunk 8]
```diff
@@ -1409,8 +1405,7 @@ def build(self):
             background_color_cell=get_color_from_hex("#451938"),
             background_color_selected_cell=get_color_from_hex("e4514f"),
         )
-        
-    .. image:: https://github.com/HeaTTheatR/KivyMD-data/raw/master/gallery/kivymddoc/data-tables-background-color-selected-cell.gif
+        .. image:: https://github.com/HeaTTheatR/KivyMD-data/raw/master/gallery/kivymddoc/data-tables-background-color-selected-cell.gif
```
Comment (HeaTTheatR) [inline kivymd/uix/datatables/datatables.py]: Removing four spaces is a bug.

File: kivymd/uix/datatables/datatables.py [hunk 9]
```diff
@@ -1383,13 +1380,12 @@ def build(self):
     .. code-block:: python
 
         self.data_tables = MDDataTable(
-            ...,
-            background_color_header=get_color_from_hex("#65275d"),
-            background_color_cell=get_color_from_hex("#451938"),
+        ...,
+        background_color_header=get_color_from_hex("#65275d"),
+        background_color_cell=get_color_from_hex("#451938"),
```
Comment (HeaTTheatR) [inline kivymd/uix/datatables/datatables.py]: Removing four spaces is a bug.

**Comment by HeaTTheatR** [comment]
@sollarp Make a new PR. I cannot trace the changes you made.

**Pull request closed without merging**